devices/usbc-cdc/usbphy-imx6ull.c

```diff
@@ -50,19 +66,56 @@ void *phy_getBase(void)
 
 u32 phy_getIrq(void)
 {
-	return 75;
+	return 75u;
 }
 
 
+static void phy_setClock(void)
+{
+	volatile u32 *ccmAnalogPllUSB0 = (u32 *)CCM_ANALOG_PLL_USB0;
+
+	imx6ull_setDevClock(clk_usboh3, 0x03);
+
+	*ccmAnalogPllUSB0 |= CCM_ANALOG_PLL_USB0_ENABLE;
+	hal_cpuDataMemoryBarrier();
+	*ccmAnalogPllUSB0 |= CCM_ANALOG_PLL_USB0_POWER;
+	hal_cpuDataMemoryBarrier();
+	while ((*ccmAnalogPllUSB0 & CCM_ANALOG_PLL_USB0_LOCK) == 0u) {
+	}
+	*ccmAnalogPllUSB0 &= ~CCM_ANALOG_PLL_USB0_BYPASS;
+	*ccmAnalogPllUSB0 |= CCM_ANALOG_PLL_USB0_EN_USB_CLKS;
+	hal_cpuDataMemoryBarrier();
+}
+
 void phy_reset(void)
 {
-	/* TODO: reset phy controller */
+	volatile u32 *usbphy = (u32 *)USB0_PHY;
+
+	*(usbphy + USB_PHY_CTRL) |= USB_PHY_CTRL_SFTRST;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_CTRL) &= ~USB_PHY_CTRL_CLKGATE;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_CTRL) &= ~USB_PHY_CTRL_SFTRST;
+	hal_cpuDataMemoryBarrier();
+	*(usbphy + USB_PHY_PWD) = 0u;
+	hal_cpuDataMemoryBarrier();
```

Comment:
This is from the imx6ull, recent official SDK, they don't use barriers in between :)
![2023-03-08-101017_1132x168_scrot](https://user-images.githubusercontent.com/141153/223671171-4176f422-963a-4e42-913c-fdf4940ee26d.png)

Reply:
Well, it certainly won't hurt anything